apps/astarte_pairing_api/test/astarte_pairing_api/agent/agent_test.exs

```diff
@@ -125,6 +125,38 @@ defmodule Astarte.Pairing.API.AgentTest do
                Agent.register_device(@test_realm, @invalid_hw_id_attrs)
     end
 
+    test "returns error changeset for negative major version in introspection" do
+      initial_introspection = %{
+        "org.astarteplatform.Values" => %{"major" => -1, "minor" => 0}
+      }
+
+      attrs = Map.put(@valid_attrs, "initial_introspection", initial_introspection)
+      assert {:error, changeset} = Agent.register_device(@test_realm, attrs)
+
+      messages =
+        Keyword.get_values(changeset.errors, :initial_introspection) |> Enum.map(&elem(&1, 0))
+
+      assert messages == [
+               "has negative versions in interface org.astarteplatform.Values"
+             ]
```

Comment:
Messages might change, as they do not depend on us but on Ecto. Imho the following check is enough for both this and the next test:
```elixir
      assert {:error, _changeset} = Agent.register_device(@test_realm, attrs)
```